doc/reference_type.xml

```diff
@@ -2,43 +2,65 @@
-	but no geometry.  What happens is that both functions are equally good to use with geometry since geometry has an autocast for both
-	-- so you end up with an ambiguous function error. To force PostgreSQL to choose, you do a CAST(mygeom As box3d)  or mygeom::box3d.</para>
+    <para>Each data type describes its type casting behaviour.
+		A <ulink url="https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-TYPE-CASTS">type cast</ulink> converts values of one data type into another type.
+		PostgreSQL allows defining casting behavior for custom types, along with the functions used to convert type values.
+		Casts can be defined as <emphasis role="bold">automatic</emphasis>.
+		This allows PostgreSQL to automatically convert a function argument to a type supported by the function.
+		Casts can also be specified <emphasis role="bold">explicitly</emphasis>,
+		using the syntax <varname>CAST(myval As sometype)</varname>  or <varname>myval::sometype</varname>.
+		This allows avoiding an issue that can occur when using an overloaded function which does not support a given type
+		(for example, one that accepts a box2d or a box3d, but not a geometry.)
+		Since geometry has an automatic cast to both box types, this will produce an "ambiguous function" error.
+	  To prevent the error use an explicit cast to the desired box type.</para>
 
-    <para><emphasis>At least as of PostgreSQL 8.3</emphasis> - Everything can be CAST to text (presumably because of the magical unknown type), so no defined CASTS for that need to be present for you to CAST an object to text.</para>
+    <para><emphasis>(At least as of PostgreSQL 8.3)</emphasis> Everything can be CAST to <varname>text</varname>, so CASTs to text do not need to be specified.</para>
     </abstract>
     </sect1info>
-    <title>PostgreSQL PostGIS Geometry/Geography/Box Types</title>
+    <title>Geometry/Geography/Box Data Types</title>
 
 	<refentry id="box2d_type">
       <refnamediv>
         <refname>box2d</refname>
-        <refpurpose>A box composed of x min, ymin,  xmax, ymax.  Often used to return the 2d enclosing box of a geometry. </refpurpose>
+        <refpurpose>A box composed of <varname>xmin, ymin, xmax, ymax</varname>.
+				Often used to describe the 2D extent of a geometry or collection of geometries. </refpurpose>
       </refnamediv>
 
       <refsection>
         <title>Description</title>
-        <para>box2d is a spatial data type used to represent the enclosing box of a geometry or set of geometries. ST_Extent in earlier versions prior to PostGIS 1.4 would return a box2d.</para>
+        <para><varname>box2d</varname> is a spatial data type used to represent
+				the two-dimensional enclosing box of a geometry or collection of geometries.
+				For example, the <varname>ST_Extent</varname> aggregate function returns a <varname>box2d</varname> object.</para>
```

Comment:
instead of wrapping into varname these can be links:
```suggestion
				For example, the <xref linkend="ST_Extent" /> aggregate function returns a <varname>box2d</varname> object.</para>
```

Reply:
Nice, definitely better.  My opinion is that everything should be linked if possible, to allow easy traversal through the docs.